### ISSUE TITLE
inconsistent handling of fields of type "uri"

### ISSUE THREAD

**Issue by u-fischer** [plk/biblatex#1304]
The following document adds more url-fields to the datamodel: a simply `urli` and a list of `urls`:

~~~~
\begin{filecontents}[overwrite]{moreurl.dbx}
\DeclareDatamodelFields[type=field, datatype=uri]{urli}
\DeclareDatamodelEntryfields{urli}
\DeclareDatamodelFields[type=list, datatype=uri]{urls}
\DeclareDatamodelEntryfields{urls}
\end{filecontents}
\begin{filecontents}[overwrite]{moreurl.bib}
@Misc{testmoreurl,
  title =   "A text with more than one url",
  author =  "Jane Doe",
  year =    "2023",
  url =     "https://www.grüße mit space.de",
  urli=     "https://www.grüße mit space.de",
  urls=     {https://www.grüße mit space.de and https://someother.de},
}
\end{filecontents}

\documentclass{article}
\usepackage[backend=biber,datamodel=moreurl,openbib]{biblatex}
\addbibresource{moreurl.bib}  
\begin{document}
Cite \cite{testmoreurl}
\end{document}
~~~~

The result in the bbl is somewhat inconsistent:

~~~~
      \verb{urlraw}
      \verb https://www.grüße mit space.de
      \endverb
      \verb{url}
      \verb https://www.xn--gre%20mit%20space-cdb84b.de
      \endverb
      \verb{urli}
      \verb https://www.grüße mit space.de
      \endverb
      \lverb{urls}{2}
      \lverb https://www.xn--gre%20mit%20space-cdb84b.de
      \lverb https://someother.de
      \endlverb
~~~~

`url` and `urls` percent encode the url, but `urli` doesn't. And only `url` has an `urlraw` variant.  Would it be possible to unify this?

**Comment by plk** [comment]
This is a bug I am looking at - just to check - those encoded URLS also look mangled too correct?

**Comment by plk** [comment]
Ah, forgot about punycode. This should all be fixed in DEV `biber` 2.20 now.

**Comment by plk** [comment]
It also needs DEV `biblatex` 3.20.
